.github/workflows/integration-testing.yml

```
@@ -124,3 +123,50 @@ jobs:
         sarif_file: rubocop.sarif
       env:
         TEST_MODE: true
+
+  test-proxy:
+    runs-on: ubuntu-latest
+    container:
+      image: ubuntu:18.04
+      options: --dns 127.0.0.1
+    services:
+      squid-proxy:
+        image: datadog/squid:latest
+        ports:
+          - 3128:3128
+    env:
+      https_proxy: http://squid-proxy:3128
+    steps:
+    - uses: actions/checkout@v2
+    - name: Move codeql-action
+      shell: bash
+      run: |
+        mkdir ../action
+        mv * .github ../action/
+        mv ../action/tests/multi-language-repo/{*,.github} .
+    - uses: ./../action/init
+      with:
+        languages: javascript
+    - uses: ./../action/analyze
+      env:
+        TEST_MODE: true
+
+  test-bypass-proxy:
+    runs-on: ubuntu-latest
+    env:
+      https_proxy: http://no-such-proxy:3128
+      no_proxy: api.github.com,github.com,github-production-release-asset-2e65be.s3.amazonaws.com
```

I'm not sure if I'm entirely happy with this value having to include `github-production-release-asset-2e65be.s3.amazonaws.com`.  This is a consequence of https://github.com/actions/http-client/blob/master/proxy.ts (which we use for downloading the codeql-bundle) not supporting any form of wildcards, so the domain has to be explicit. Maybe this is fine if this domain is for all of github, but I'm worried about it changing when we update the bundle, or just at random.

The options that I see at the moment are to just not include this test, or to go with this hardcoded value and raise a PR on https://github.com/actions/http-client to improve support there.

@chrisgavin or @tibbes, do you have any thoughts here?

---

It does seem a little bit dodgy.

Perhaps another option is to use [mitmproxy](https://mitmproxy.org/) with a script that blocks everything except S3 URLs.

---

After more thought and trying to get this working, I think in fact it's more fair to say that we only partially support `no_proxy`. As far as I can tell there isn't really a format spec for this, but more a general convention between implementations, and the one we're using seems to be on the minimal side.

I'm going to remove this test and instead open an issue to improve this support and add the test back in.